### PR TITLE
Initial Node drain docs

### PR DESCRIPTION
**Pull request by schmichael**
More incoming!

### CODE REVIEW

File: website/source/docs/commands/node/drain.html.md.erb [hunk 1]
```diff
@@ -37,6 +39,19 @@ operation is desired.
...
 * `-enable`: Enable node drain mode.
 * `-disable`: Disable node drain mode.
+* `-deadline`: Set the deadline by which all allocations must be moved off the
+  node. Remaining allocations after the deadline are forced removed from the
+  node.
+* `-detach`: Return immediately instead of entering monitor mode.
+* `-force`: Force remove allocations off the node immediately.
+* `-no-deadline`: No deadline allows the allocations to drain off the node
+  without being force stopped after a certain deadline.
+* `-ignore-system`: Ignore ssytem allows the drain to complete without stopping
```
Comment (lorenzo) [inline website/source/docs/commands/node/drain.html.md.erb]: Typo: ssystem

Comment (schmichael) [reply website/source/docs/commands/node/drain.html.md.erb]: Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,15 @@
 ## 0.8 (Unreleased)
 
 __BACKWARDS INCOMPATIBILITIES:__
+ * cli: node-drain now blocks until the drain completes and all allocations on
```
Comment (dadgar) [inline CHANGELOG.md]: Use new command format

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+   the draining node have stopped. Use -detach for the old behavior.
  * discovery: Prevent absolute URLs in check paths. The documentation indicated
    that absolute URLs are not allowed, but it was not enforced. Absolute URLs
    in HTTP check paths will now fail to validate. [[GH-3685](https://github.com/hashicorp/nomad/issues/3685)]
+ * drain: Draining a node no longer stops all allocations immediately: a new
+   [migrate stanza](https://www.nomadproject.io/docs/job-specification/migrate.html)
+   allows jobs to specify how quickly task groups can be drained. A `-force`
+   option was added to emulate the old drain behavior.
```
Comment (dadgar) [inline CHANGELOG.md]: "was added to" -> "can be used to"

File: CHANGELOG.md [hunk 4]
```diff
@@ -21,6 +27,9 @@ IMPROVEMENTS:
  * core: Servers can now retry connecting to Vault to verify tokens without requiring a SIGHUP to do so [[GH-3957](https://github.com/hashicorp/nomad/issues/3957)]
  * core: Updated yamux library to pick up memory and CPU performance improvements [[GH-3980](https://github.com/hashicorp/nomad/issues/3980)]
  * core: Client stanza now supports overriding total memory [[GH-4052](https://github.com/hashicorp/nomad/issues/4052)]
+ * core: Node draining is now able to migrate allocations in a controlled
+   manner with parameters specified by the drain command and in job files.
```
Comment (dadgar) [inline CHANGELOG.md]: in job files using the migrate stanza

File: command/job_init.go [hunk 5]
```diff
@@ -159,6 +159,35 @@ job "example" {
     canary = 0
   }
 
+  # The migrate stanza specifies the group's strategy for migrating off of
+  # draining nodes. If omitted, a default migration strategy is applied.
+  #
+  # For more information on the "migrate" stanza, please see 
+  # the online documentation at:
+  #
+  #     https://www.nomadproject.io/docs/job-specification/migrate.html
```
Comment (dadgar) [inline command/job_init.go]: Put in the group?

File: command/job_init.go [hunk 6]
```diff
@@ -159,6 +159,35 @@ job "example" {
...
+  #
+  migrate {
+    # Specifies the number of task groups that can be migrated at the same
+    # time. This number must be less than the total count for the group as
+    # (count - max_parallel) will be left running during migrations.
+    max_parallel     = 1
```
Comment (dadgar) [inline command/job_init.go]: Why are the equal signs so far off

File: website/source/docs/commands/node/eligibility.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,68 @@
+---
```
Comment (dadgar) [inline website/source/docs/commands/node/eligibility.html.md.erb]: Add link the the commands/node.html.md.erb file

File: website/source/docs/commands/node/drain.html.md.erb [hunk 8]
```diff
@@ -10,7 +10,20 @@ description: >
...
+migrating all existing allocations away. Allocations will be migrated according
+to their [`migrate`][migrate] stanza until the drain's deadline is reached.
+
+By default the `node drain` command blocks until a node is done draining and
+all allocations have terminated. Canceling the `node drain` command *will not*
+cancel the drain. Drains may be canceled by using the `-disable` parameter
+below.
+
+When draining more than one node at a time, it is recommended you first disable
+[scheduling eligibility][eligibility] on all nodes you are about to drain. For
```
Comment (dadgar) [inline website/source/docs/commands/node/drain.html.md.erb]: "you are about to drain" -> "that will be drained"

File: website/source/docs/commands/node/eligibility.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+layout: "docs"
+page_title: "Commands: node eligibility"
+sidebar_current: "docs-commands-node-eligibility"
+description: >
+  The node drain command is used to configure a node's scheduling eligibility.
```
Comment (dadgar) [inline website/source/docs/commands/node/eligibility.html.md.erb]: Referencing drain not eligiblity

File: website/source/docs/commands/node/drain.html.md.erb [hunk 10]
```diff
@@ -53,3 +79,21 @@ Enable drain mode on the local node:
 ```
 $ nomad node drain -enable -self
 ```
+
```
Comment (dadgar) [inline website/source/docs/commands/node/drain.html.md.erb]: Display some sample output?

File: website/source/docs/commands/node/eligibility.html.md.erb [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+* `-disable`: Disable scheduling eligibility.
+* `-self`: Set eligibility for the local node.
+* `-yes`: Automatic yes to prompts.
+
+## Examples
+
+Enable scheduling eligibility on node with ID prefix "4d2ba53b":
+
+```
+$ nomad node eligibility -enable 4d2ba53b
```
Comment (dadgar) [inline website/source/docs/commands/node/eligibility.html.md.erb]: Display output?

File: website/source/docs/job-specification/migrate.html.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  migrate {
+    max_parallel     = 1
+    health_check     = "checks"
+    min_healthy_time = "10s"
+    healthy_deadline = "5m"
+  }
+}
+```
+
+When one or more nodes are draining, only `migrate.max_parallel` allocations
```
Comment (dadgar) [inline website/source/docs/job-specification/migrate.html.md]: do we need the migrate. prefix? It is the doc page for this stanza

File: website/source/docs/job-specification/migrate.html.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+`migrate.healthy_deadline` is reached.
+
+Note that a node's drain [deadline][deadline] will override the `migrate`
+stanza for allocations on that node. The `migrate` stanza is for job authors to
+define how their services should be migrated, while the node drain deadline is
+for system operators to put hard limits on how long a drain may take.
+
+## `migrate` Parameters
+
+- `max_parallel` `(int: 1)` - Specifies the number of task groups that can be
```
Comment (dadgar) [inline website/source/docs/job-specification/migrate.html.md]: task groups > allocations

File: website/source/docs/job-specification/migrate.html.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+  migrated at the same time. This number must be less than the total
+  [`count`][count] for the group as `count - max_parallel` will be left running
+  during migrations.
+
+- `health_check` `(string: "checks")` - Specifies the mechanism in which
+  allocations health is determined. The potential values are:
+
+  - "checks" - Specifies that the allocation should be considered healthy when
+    all of its tasks are running and their associated [checks][] are healthy,
```
Comment (dadgar) [inline website/source/docs/job-specification/migrate.html.md]: Missing link for checks

File: website/source/docs/job-specification/migrate.html.md (hunk 12 above)
Comment (schmichael) [reply website/source/docs/job-specification/migrate.html.md]: yeah, it looks gross and seems useless. removing.

File: website/source/docs/commands/node/drain.html.md.erb [hunk 15]
```diff
@@ -37,6 +50,19 @@ operation is desired.
 
 * `-enable`: Enable node drain mode.
 * `-disable`: Disable node drain mode.
+* `-deadline`: Set the deadline by which all allocations must be moved off the
+  node. Remaining allocations after the deadline are forced removed from the
```
Comment (dadgar) [inline website/source/docs/commands/node/drain.html.md.erb]: force removed

File: website/source/docs/commands/node/drain.html.md.erb [hunk 16]
```diff
@@ -37,6 +50,19 @@ operation is desired.
 
 * `-enable`: Enable node drain mode.
 * `-disable`: Disable node drain mode.
+* `-deadline`: Set the deadline by which all allocations must be moved off the
```
Comment (dadgar) [inline website/source/docs/commands/node/drain.html.md.erb]: Mention the default of 1h

File: command/job_init.go (hunk 5 above)
Comment (schmichael) [reply command/job_init.go]: Left it for now to match update

File: website/source/docs/commands/node/eligibility.html.md.erb (hunk 7 above)
Comment (schmichael) [reply website/source/docs/commands/node/eligibility.html.md.erb]: done

**Pull request merged**